src/lib.rs

```diff
@@ -181,6 +184,26 @@ impl<'a> fmt::Display for Record<'a> {
     }
 }
 
+/// Write Record Extension on io::Write
+pub trait WriteRecordExt: io::Write {
+    
+    /// write a record to a io::Write implementor
+    ///
+    /// returns the length of the written record
+    fn write_record(&mut self, record: Record) -> Result<usize>;
```

Comment:
Returning length here is redundant since it is using `io::Write::write_all` internally which is follows *write all or error* logic, so user always know the amount of bytes written (via `record.as_ref().len()`).
I think return type of `WriteRecordExt::write_record` should follow the return type of `io::Write::write_all` which is `Result<()>`